File: parse_md.js
import fs from 'fs';
// Removed uuid dependency

const filePath = 'c:/Users/kyle/Downloads/Schafer_Family_Cookbook_Complete.md';
const outputPath = 'c:/Users/kyle/Downloads/SchaferFamilyCookbook/src/data/recipes.json';

try {
    const content = fs.readFileSync(filePath, 'utf-8');
    const lines = content.split('\n');

    const recipes = [];
    let currentRecipe = null;
    let section = null;

    for (let line of lines) {
        line = line.trim();
        if (!line || line.startsWith('---')) continue;

        // Detect Header
        if (line.startsWith('### ')) {
            if (currentRecipe) recipes.push(currentRecipe);

            const title = line.replace('### ', '').trim();
            if (['The Oehler Family', 'The Schafer Family'].includes(title)) {
                currentRecipe = null;
                continue;
            }

            currentRecipe = {
                id: 'imported_' + Math.random().toString(36).substr(2, 9),
                title: title,
                contributor: "Family",
                category: "Main",
                ingredients: [],
                instructions: [],
                image: "https://images.unsplash.com/photo-1495195129352-aec325a55b65?auto=format&fit=crop&q=80&w=800",
                notes: ""
            };
            section = null;
            continue;
        }

        if (!currentRecipe) continue;

        if (line.startsWith('**Contributor:**')) {
            currentRecipe.contributor = line.replace('**Contributor:**', '').trim();
            continue;
        }
        if (line.startsWith('**Ingredients:**')) {
            section = 'ingredients';
            continue;
        }
        if (line.startsWith('**Instructions:**')) {
            section = 'instructions';
            continue;
        }
        if (line.startsWith('**Notes:**')) {
            section = 'notes';
            currentRecipe.notes = line.replace('**Notes:**', '').trim();
            continue;
        }

        if (section === 'ingredients' && line.startsWith('- ')) {
            currentRecipe.ingredients.push(line.replace('- ', ''));
        } else if (section === 'instructions') {
            const clean = line.replace(/^\d+\.\s*/, '');
            if (clean) currentRecipe.instructions.push(clean);
        } else if (section === 'notes') {
            currentRecipe.notes += " " + line;
        }
    }

    if (currentRecipe) recipes.push(currentRecipe);

    const validRecipes = recipes.filter(r => r.ingredients.length > 0);

    // Categorize
    validRecipes.forEach(r => {
        const t = r.title.toLowerCase();
        if (t.includes('cookie') || t.includes('bar') || t.includes('cake') || t.includes('brownie') || t.includes('pie') || t.includes('fudge') || t.includes('dessert') || t.includes('chow')) {
            r.category = 'Dessert';
        } else if (t.includes('soup') || t.includes('stew') || t.includes('chili')) {
            r.category = 'Main';
        } else if (t.includes('salad') && !t.includes('chicken')) {
            r.category = 'Side';
        } else if (t.includes('dip') || t.includes('salsa')) {
            r.category = 'Dip/Sauce';
        } else if (t.includes('bread')) {
            r.category = 'Bread';
        } else if (t.includes('breakfast') || t.includes('egg') || t.includes('pancake')) {
            r.category = 'Breakfast';
        } else if (t.includes('snack') || t.includes('popcorn')) {
            r.category = 'Snack';
        }
    });

    fs.writeFileSync(outputPath, JSON.stringify(validRecipes, null, 2));
    console.log(`Successfully converted ${validRecipes.length} recipes.`);

} catch (e) {
    console.error(e);
}

File: parse_md.py
import re
import json
import uuid

def parse_recipe_markdown(file_path):
    with open(file_path, 'r', encoding='utf-8') as f:
        lines = f.readlines()

    recipes = []
    current_recipe = None
    section = None
    
    # Simple state machine
    for line in lines:
        line = line.strip()
        if not line or line.startswith('---'):
            continue
            
        # Detect Header (Recipe Title)
        if line.startswith('### '):
            if current_recipe:
                recipes.append(current_recipe)
            
            title = line.replace('### ', '').strip()
            # Skip non-recipe headers
            if title in ['The Oehler Family', 'The Schafer Family', 'Festive Apple Dip']: 
                # Festive Apple Dip IS a recipe, but 'The Oehler Family' isn't
                pass
            
            current_recipe = {
                "id": str(uuid.uuid4())[:8],
                "title": title,
                "contributor": "Family",
                "category": "Main", # Default, will map later if possible or just manual
                "ingredients": [],
                "instructions": [],
                "image": "https://images.unsplash.com/photo-1495195129352-aec325a55b65?auto=format&fit=crop&q=80&w=800", # Generic
                "notes": ""
            }
            section = None
            continue

        if not current_recipe:
            continue

        # Detect Metadata
        if line.startswith('**Contributor:**'):
            current_recipe['contributor'] = line.replace('**Contributor:**', '').strip()
            continue

        if line.startswith('**Ingredients:**'):
            section = 'ingredients'
            continue

        if line.startswith('**Instructions:**'):
            section = 'instructions'
            continue

        if line.startswith('**Notes:**'):
            section = 'notes'
            current_recipe['notes'] = line.replace('**Notes:**', '').strip()
            continue

        # Content Parsing
        if section == 'ingredients':
            if line.startswith('- '):
                current_recipe['ingredients'].append(line.replace('- ', ''))
        
        elif section == 'instructions':
            # Remove "1. ", "2. ", etc
            clean = re.sub(r'^\d+\.\s*', '', line)
            if clean:
                current_recipe['instructions'].append(clean)
                
        elif section == 'notes':
            current_recipe['notes'] += " " + line

    if current_recipe:
        recipes.append(current_recipe)

    # Filter out non-recipes (heuristics)
    valid_recipes = [r for r in recipes if r['ingredients']]
    
    # Categorize (Basic Heuristic)
    for r in valid_recipes:
        t = r['title'].lower()
        if 'cookie' in t or 'bar' in t or 'cake' in t or 'brownie' in t or 'pie' in t or 'fudge' in t or 'dessert' in t or 'chow' in t:
             r['category'] = 'Dessert'
        elif 'soup' in t or 'stew' in t or 'chili' in t:
             r['category'] = 'Main' # technically soup/appetizer, but Main is safe
        elif 'salad' in t and 'chicken' not in t:
             r['category'] = 'Side'
        elif 'dip' in t or 'salsa' in t:
             r['category'] = 'Dip/Sauce'
        elif 'bread' in t:
             r['category'] = 'Bread'
        elif 'breakfast' in t or 'egg' in t or 'pancake' in t:
             r['category'] = 'Breakfast'
        elif 'snack' in t or 'popcorn' in t:
             r['category'] = 'Snack'

    return valid_recipes

recipes = parse_recipe_markdown('c:/Users/kyle/Downloads/Schafer_Family_Cookbook_Complete.md')
with open('c:/Users/kyle/Downloads/SchaferFamilyCookbook/src/data/recipes.json', 'w', encoding='utf-8') as f:
    json.dump(recipes, f, indent=2)

print(f"Successfully converted {len(recipes)} recipes to JSON.")
